src/core/tests/float8_e5m2.cpp

```diff
@@ -30,12 +30,145 @@ TEST(F8E5M2Test, f32_inf) {
+}
+
+TEST(F8E5M2Test, f8e5m2_num_limits_digits) {
+    EXPECT_EQ(std::numeric_limits<ov::float8_e5m2>::digits, 3);
+}
+
+TEST(F8E5M2Test, f8e5m2_num_limits_digits10) {
+    const auto f8_dig = std::numeric_limits<ov::float8_e5m2>::digits;
+    const auto f8_dig10 = std::numeric_limits<ov::float8_e5m2>::digits10;
+
+    EXPECT_EQ(f8_dig10, static_cast<int>((f8_dig - 1) * std::log10(2)));
+    EXPECT_EQ(f8_dig10, 0);
+}
+
+TEST(F8E5M2Test, f8e5m2_num_limits_epsilon) {
+    const auto f8_1 = ov::float8_e5m2(1.f);
+    const auto f8_1_bits = f8_1.to_bits();
+    const auto f8_1_next_bits = f8_1_bits + 1u;
+
+    const auto f8_eps = ov::float8_e5m2::from_bits(f8_1_next_bits - f8_1_bits);
+
+    EXPECT_EQ(f8_eps, std::numeric_limits<ov::float8_e5m2>::epsilon());
+    EXPECT_EQ(f8_eps.to_bits(), std::numeric_limits<ov::float8_e5m2>::epsilon().to_bits());
+}
+
+TEST(F8E5M2Test, f8e5m2_num_limits_round_error) {
+    const auto f8 = ov::float8_e5m2(0.5f);
+
+    EXPECT_EQ(f8, std::numeric_limits<ov::float8_e5m2>::round_error());
+    EXPECT_EQ(f8.to_bits(), std::numeric_limits<ov::float8_e5m2>::round_error().to_bits());
+}
+
+TEST(F8E5M2Test, f8e5m2_quiet_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<ov::float8_e5m2>::quiet_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b01111111);
+    EXPECT_TRUE(std::numeric_limits<ov::float8_e5m2>::has_quiet_NaN);
+    EXPECT_EQ(std::numeric_limits<ov::float8_e5m2>::quiet_NaN().to_bits(), 0b01111111);
+}
+
+TEST(F8E5M2Test, f8e5m2_sig_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<ov::float8_e5m2>::signaling_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b01111101);
+    EXPECT_TRUE(std::numeric_limits<ov::float8_e5m2>::has_signaling_NaN);
+    EXPECT_EQ(std::numeric_limits<ov::float8_e5m2>::signaling_NaN().to_bits(), 0b01111101);
+}
+
+TEST(F8E5M2Test, f16_quiet_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<ov::float16>::quiet_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b01111111);
+    EXPECT_TRUE(std::numeric_limits<ov::float8_e5m2>::has_quiet_NaN);
+    EXPECT_EQ(std::numeric_limits<ov::float8_e5m2>::quiet_NaN().to_bits(), 0b01111111);
+}
+
+TEST(F8E5M2Test, f16_sig_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<ov::float16>::signaling_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b01111101);
+    EXPECT_TRUE(std::numeric_limits<ov::float8_e5m2>::has_signaling_NaN);
+    EXPECT_EQ(std::numeric_limits<ov::float8_e5m2>::signaling_NaN().to_bits(), 0b01111101);
+}
+
+TEST(F8E5M2Test, f32_quiet_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<float>::quiet_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    // The last bit is zero because of f32 to f16 conversion
+    EXPECT_EQ(f8.to_bits(), 0b01111110);
+}
+
+TEST(F8E5M2Test, f32_sig_nan) {
+    const auto f8 = ov::float8_e5m2(std::numeric_limits<float>::signaling_NaN());
```

Comment:
So if this conversion is supported the test table should be updated as there is only quiet NaN.

Reply:
Added signaling NaN to the test values table